crates/turbopack-tests/tests/execution.rs

```diff
@@ -0,0 +1,345 @@
+// skip.
+#[testing::fixture("tests/execution/*/*/*", exclude("node_modules|__skipped__"))]
+fn test(resource: PathBuf) {
+    if resource.ends_with("__skipped__") {
+        return;
+    }
+
+    let messages = get_messages(run(resource, IssueSnapshotMode::Snapshots).unwrap());
+    if !messages.is_empty() {
+        panic!(
+            "Failed with error(s) in the following test(s):\n\n{}",
+            messages.join("\n\n--\n")
+        )
+    }
+}
+
+#[testing::fixture("tests/execution/*/*/__skipped__/*/input")]
+#[should_panic]
+fn test_skipped_fails(resource: PathBuf) {
+    let resource = resource.parent().unwrap().to_path_buf();
+
+    let JsResult {
+        // Ignore uncaught exceptions for skipped tests.
+        uncaught_exceptions: _,
+        unhandled_rejections: _,
+        jest_result,
+    } = run(resource, IssueSnapshotMode::NoSnapshots).unwrap();
+
+    // Assert that this skipped test itself has at least one browser test which
+    // fails.
+    assert!(
+        // Skipped tests sometimes have errors (e.g. unsupported syntax) that prevent tests from
+        // running at all. Allow them to have empty results.
+        jest_result.test_results.is_empty()
+            || jest_result
+                .test_results
+                .into_iter()
+                .any(|r| !r.errors.is_empty()),
+    );
+}
+
+fn get_messages(js_results: JsResult) -> Vec<String> {
+    let mut messages = vec![];
+
+    if js_results.jest_result.test_results.is_empty() {
+        messages.push("No tests were run.".to_string());
+    }
+
+    for test_result in js_results.jest_result.test_results {
+        // It's possible to fail multiple tests across these tests,
+        // so collect them and fail the respective test in Rust with
+        // an aggregate message.
+        if !test_result.errors.is_empty() {
+            messages.push(format!(
+                "\"{}\":\n{}",
+                test_result.test_path[1..].join(" > "),
+                test_result.errors.join("\n")
+            ));
+        }
+    }
+
+    for uncaught_exception in js_results.uncaught_exceptions {
+        messages.push(format!("Uncaught exception: {}", uncaught_exception));
+    }
+
+    for unhandled_rejection in js_results.unhandled_rejections {
+        messages.push(format!("Unhandled rejection: {}", unhandled_rejection));
+    }
+
+    messages
+}
+
+#[tokio::main(flavor = "current_thread")]
+async fn run(resource: PathBuf, snapshots: IssueSnapshotMode) -> Result<JsResult> {
+    register();
+
+    let tt = TurboTasks::new(MemoryBackend::default());
+    tt.run_once(async move {
+        let resource_str = resource.to_str().unwrap();
+        Ok(match snapshots {
+            IssueSnapshotMode::Snapshots => {
+                (*run_test_and_snapshot_issues(resource_str).await.unwrap()).clone()
+            }
+            IssueSnapshotMode::NoSnapshots => {
+                (*run_test(resource_str).await?.js_result.await?).clone()
+            }
+        })
```

Comment:
Nit (pseudo code, please fix)
```suggestion
        let run_result = run_test(resource);
        if matches!(snapshots, IssueSnapshotMode::Snapshots) {
          // snapshots
        }
```